relay/writer/file.go

```diff
@@ -216,43 +216,57 @@ func (w *FileWriter) handleFormatDescriptionEvent(ev *replication.BinlogEvent) (
 // handle RotateEvent:
 //   1. update binlog filename if needed
 //   2. write the RotateEvent if not fake
+// NOTE: we only see fake event for RotateEvent in MySQL source code,
+//       if see fake event for other event type, then handle them.
```

Comment:
shall we add some log if `handlePotentialHoleOrDuplicate` in not expected event type handler

Reply:
add some log for `handleFileHoleExist` and `handleDuplicateEventsExist`